gettsim/taxes/zve.py

```diff
@@ -312,58 +316,151 @@ def calc_gross_e7(tax_unit, params):
+            )
+        else:
+            # No "vorwegab" anymore after 2019.
+            item_1 = 0
+        # calcuate the remaining amount.
+        vorsorg_rest = np.maximum(
+            12 * (tax_unit["rvbeit"] + tax_unit["gkvbeit"]) - item_1, 0
+        )
+        # Deduct a 'Grundhöchstbetrag' (1334€ in 2004),
+        # or the actual expenses if lower (which is unlikely)
+        item_2 = np.minimum(params["grundbet"], vorsorg_rest)
+        # From what is left from vorsorg_rest, you may deduct 50%.
+        # (up until 50% of 'Grundhöchstbetrag')
+        item_3 = np.minimum(0.5 * (vorsorg_rest - item_2), 0.5 * params["grundbet"])
+    # For the married couple, the same stuff, but with tu totals.
+    if tax_unit["zveranl"].max():
+        for var in ["m_wage", "rvbeit", "gkvbeit"]:
+            tax_unit[f"{var}_tu"] = tax_unit.loc[~tax_unit["child"], "m_wage"].sum()
+        if params["year"] <= 2019:
+            item_1 = 0.5 * np.maximum(
+                2 * params["vorwegab"]
+                - params["kuerzquo"] * 12 * tax_unit["m_wage_tu"],
+                0,
+            )
+        else:
+            item_1 = 0
+
+        vorsorg_rest = 0.5 * np.maximum(
+            12 * (tax_unit["rvbeit_tu"] + tax_unit["gkvbeit_tu"]) - item_1, 0
+        )
+        item_2 = 0.5 * np.minimum(params["grundbet"], vorsorg_rest)
+        item_3 = 0.5 * np.minimum((vorsorg_rest - item_2), 2 * params["grundbet"])
+
+    # Finally, add up all three amounts and assign it to the adults.
+    vorsorge = ~tax_unit["child"] * (item_1 + item_2 + item_3).astype(int)
+    # The test requires a named item. If there is only one value, create a list.
+    return vorsorge
+
+
+def vorsorge_since_2005(tax_unit, params, soz_vers_beitr_params):
+    """ With the 2005 reform, no taxpayer was supposed to be affected negatively.
+        Therefore, one needs to compute amounts
+        (2004 and 2005 regime) and take the higher one.
+    """
+    vors2004 = vorsorge_pre_2005(tax_unit, params, soz_vers_beitr_params)
+    vors2005 = _vorsorge_since_2005(tax_unit, params, soz_vers_beitr_params)
+
+    return np.maximum(vors2004, vors2005)
 
-def vorsorge_dummy(tax_unit, params, soz_vers_beitr_params):
-    return 0
+
+def vorsorge_since_2010(tax_unit, params, soz_vers_beitr_params):
+    """ After a supreme court ruling, the 2005 rule had to be changed in 2010.
+        Therefore, one needs to compute amounts
+        (2004 and 2010 regime) and take the higher one. (§10 (3a) EStG).
+        Sidenote: The 2010 ruling is by construction
+        *always* more or equally beneficial than the 2005 one,
+        so no need for a separate check there.
+    """
+    vors2004 = vorsorge_pre_2005(tax_unit, params, soz_vers_beitr_params)
+    vors2010 = _vorsorge_since_2010(tax_unit, params, soz_vers_beitr_params)
+
+    return np.maximum(vors2004, vors2010)
+
+
+def calc_altersvors_aufwend(tax_unit, params):
```

Comment:
Something more general: As we use german words for functions, shouldn't we replace calc everywhere?

Reply:
maybe worth a discussion point in #60?

Reply:
I would just get rid of `calc_`. The only case where this might matter is where we have assignments like:

```python
altersvors_aufwend = calc_altersvors_aufw(...)
```
but this does not seem worth adding a nonsense string to every function.

Thinking of it, another option would be to only ever import functions as follows: 

```
import gettsim.taxes as tx
import gettsim.transfers as tr
```

and get rid of the intermediate module level by `__init__.py` files in `gettsim.taxes` and `gettsim.transfers` (these files would only include things like `from .zve import altersvors_aufwend`). So the above example would become:

```python
import gettsim.taxes as tx

altersvors_aufwend = tx.altersvors_aufw(...)
```

I kind of like that idea (the Zen... **Namespaces are one honking great idea -- let's do more of those!**)